jans-auth-server/server/src/main/java/io/jans/as/server/audit/debug/ServletLoggingFilter.java

```diff
@@ -119,10 +122,17 @@ protected String getRequestDescription(RequestWrapper requestWrapper, Duration d
         }
     }
 
-    protected String getResponseDescription(ResponseWrapper responseWrapper) {
+
+    protected String getResponseDescription(HttpServletResponseWrapper responseWrapper) {
         HttpResponse httpResponse = new HttpResponse();
-        httpResponse.setStatus(responseWrapper.getStatus());
-        httpResponse.setHeaders(responseWrapper.getHeaders());
+        if (appConfiguration.getHttpLoggingResponseBodyContent()) {
+            httpResponse.setStatus(((ReadableResponseWrapper) responseWrapper).getStatus());
+            httpResponse.setHeaders(((ReadableResponseWrapper) responseWrapper).getHeaders());
+            httpResponse.setBody(((ReadableResponseWrapper) responseWrapper).readBodyValue());
```

Comment:
@jmunozherbas would you please extensively test this change and confirm it works good ? Because this line sounds like we read from writer and close it. So down the chain filters will not be able to process it. As result it can cause not returing response out of the server. It's similar problem as reading input stream in request which can make it unavailable further.

See this https://stackoverflow.com/questions/10210645/http-servlet-request-lose-params-from-post-body-after-read-it-once

Reply:
I have applied some changes.
I have tested with jans-auth-client tests and results are similar than before the change.
Clarification.- The changes and new flows just be applied if the parameter "httpLoggingResponseBodyContent" is enabled, if this parameter doesn´t exist or is false, then the flow is executed as usual.

The code to read body from response is based on next example:

search "Logging requests and responses" in next page:
https://spaces.at.internet2.edu/display/Grouper/Grouper+Web+Services

code repository:
https://github.com/Internet2/grouper/tree/master/grouper-ws/grouper-ws/src/grouper-ws/edu/internet2/middleware/grouper/ws/j2ee

maven repository of this code library:
https://mvnrepository.com/artifact/edu.internet2.middleware.grouper/grouper-ws/5.0.3

There is an example to read request body too, maybe could be implemented.
The log printting in just one line (request and response) could be implemented too, similar to the example code.

Reply:
If all tests are passed with `httpLoggingResponseBodyContent=true` then I guess we are good to merge.